telescope/selector.py

```diff
@@ -186,64 +185,35 @@ def parse_ip_translation(self, ip_translation_dict):
 
   def validate_selector_input(self, selector_dict):
-    if not selector_dict.has_key('file_format_version') or \
-            selector_dict['file_format_version'] < self.supported_file_format_versions['minimum'] or \
-            selector_dict['file_format_version'] > self.supported_file_format_versions['maximum']:
-      raise ValueError('UnsupportedSelectorVersion')
-
-    if not selector_dict.has_key('duration'):
-      raise ValueError('UnsupportedDuration')
-
-    if not selector_dict.has_key('metric') or \
-            (type(selector_dict['metric']) != str and type(selector_dict['metric']) != unicode) or \
-            (selector_dict['metric'] not in self.supported_metrics and \
-             selector_dict['metric'] != 'all'):
-      raise ValueError('UnsupportedMetric')
-
-    """ For now we only support subset specifications with 1 or 2 (isp, site,
-        client) tuples
-    """
-    if len(selector_dict['subsets']) > 2 or len(selector_dict['subsets']) < 1:
-        raise ValueError('UnsupportedSubsetSize')
-
-    if not selector_dict.has_key('subsets') or \
-            type(selector_dict['subsets']) != list:
-      raise ValueError('UnsupportedSubsets')
-
-    for tuple_set in selector_dict['subsets']:
-      if sorted(tuple_set.keys()) != sorted(self.supported_subset_keys):
-        raise ValueError('UnsupportedSubsetDefinition')
-
-    """ Selectors should contained two control variables and one independendent
-        variable
-    """
-    if len(selector_dict['subsets']) == 2:
-      self.find_independent_variable(selector_dict['subsets'])
+    if not selector_dict.has_key('file_format_version'):
+        raise ValueError('NoSelectorVersionSpecified')
+    elif selector_dict['file_format_version'] == 1.0:
+        raise ValueError('UnsupportedVersion')
+    elif selector_dict['file_format_version'] == 1.1:
+        parser_rules = SelectorFileParserRules1_1()
+    else:
+        raise ValueError('UnsupportedSelectorVersion')
 
+    parser_rules.parse(selector_dict)
     return True
 
-  def find_independent_variable(self, subsets):
-    """ Parse two (isp, site, timestamp) tuples and return the key of the
-        independent variable.
-
-        Args:
-          subsets (list): List of length 2 with (isp, site, timestamp) dicts.
-
-        Returns:
-          str: name of the key that is different between the two tuples.
-
-    """
-
-    independent_variable = None
-
-    for key in subsets[0].keys():
-      if subsets[0][key] != subsets[1][key] and independent_variable is None:
-        independent_variable = key
-      elif subsets[0][key] != subsets[1][key] and \
-          independent_variable is not None:
-        raise ValueError('IncomparableSets')
-
-    if independent_variable == None:
-      raise Exception('NoIndependentVariable')
-
-    return independent_variable
+class SelectorFileParserRules(object):
+    def parse(self, selector_dict):
```

Comment:
Don't think we should call it `parse` because it's not parsing. `validate`? `check_rules`?